Import res.partner.category

Tested on Odoo 7.0 and Magento 1.7.0.0 (with SAMPLE_DATA) 

Synchronization works well, the import of res.partner.category works, except for categories with a compound name (eg "TEST GROUP") 
I get the following traceback:

    Traceback (most recent call last):
      File "/home/clovis/Documents/test/connector/connector/queue/worker.py", line 122, in run_job
        job.perform(session)
      File "/home/clovis/Documents/test/connector/connector/queue/job.py", line 491, in perform
        self.result = self.func(session, *self.args, **self.kwargs)
      File "/home/clovis/Documents/test/connector-magento/magentoerpconnect/unit/import_synchronizer.py", line 378, in import_record
        importer.run(magento_id, force=force)
      File "/home/clovis/Documents/test/connector-magento/magentoerpconnect/unit/import_synchronizer.py", line 232, in run
        self.binder.bind(self.magento_id, binding_id)
      File "/home/clovis/Documents/test/connector-magento/magentoerpconnect/unit/binder.py", line 125, in bind
        "got: %s, %s" % (external_id, binding_id)
    AssertionError: external_id or binding_id missing, got: 0, 17

I am also running into this issue with OpenERP 7.0 and Magento ver. 1.8.1.0 and I get this traceback:

```
Traceback (most recent call last):
  File "/opt/openerp/addons/connector/connector/queue/worker.py", line 122, in run_job
    job.perform(session)
  File "/opt/openerp/addons/connector/connector/queue/job.py", line 491, in perform
    self.result = self.func(session, *self.args, **self.kwargs)
  File "/opt/openerp/addons/connector-magento/magentoerpconnect/unit/import_synchronizer.py", line 378, in import_record
    importer.run(magento_id, force=force)
  File "/opt/openerp/addons/connector-magento/magentoerpconnect/unit/import_synchronizer.py", line 232, in run
    self.binder.bind(self.magento_id, binding_id)
  File "/opt/openerp/addons/connector-magento/magentoerpconnect/unit/binder.py", line 125, in bind
    "got: %s, %s" % (external_id, binding_id)
AssertionError: external_id or binding_id missing, got: 0, 6
```

--- a/magentoerpconnect/unit/binder.py
+++ b/magentoerpconnect/unit/binder.py
@@ -120,7 +120,9 @@ class MagentoModelBinder(MagentoBinder):
         context = self.session.context.copy()
         context['connector_no_export'] = True
         now_fmt = datetime.now().strftime(DEFAULT_SERVER_DATETIME_FORMAT)
-        assert external_id and binding_id, (
+        # the external ID can be 0 on Magento! Prevent False values
+        # like False, None, or "", but not 0.
+        assert (external_id or external_id == 0) and binding_id, (
             "external_id or binding_id missing, "
             "got: %s, %s" % (external_id, binding_id)
         )

